lib/kitchen/driver/vcenter.rb

```diff
@@ -337,7 +337,15 @@ def get_host(name, datacenter, cluster = nil)
       # @param [type] type is the type of the folder, one of VIRTUAL_MACHINE, DATACENTER, possibly other values
       def get_folder(name, type = "VIRTUAL_MACHINE")
         folder_api = VSphereAutomation::VCenter::FolderApi.new(api_client)
-        folders = folder_api.list({ filter_names: name, filter_type: type }).value
+
+        # Handle sub-folders
+        if name.include? "/"
+          folder_stack = name.split("/").reject(&:empty?)
+          name = folder_stack.pop
+          folders = folder_api.list({ parent_folders: folder_stack, filter_names: name, filter_type: type }).value
+        else
+          folders = folder_api.list({ filter_names: name, filter_type: type }).value
+        end
```

Comment:
Hi,

* First, looking at the docs for this API, it looks like you've used the wrong key name here, it should be `:filter_parent_folders`: https://vmware.github.io/vsphere-automation-sdk-ruby/VSphereAutomation/VCenter/FolderApi.html
* There is also some repetition here that could be simplified.

```suggestion
        options = { filter_names: name, filter_type: type }
        if name.include? "/"
          folder_stack = name.split("/").reject(&:empty?)
          name = folder_stack.pop
          options[:filter_names] = name
          options[:filter_parent_folders] = folder_stack
        end
        folders = folder_api.list(options).value
```

Reply:
Really like the suggestion, thank you! I have adopted your idea and added a new commit.

Reply:
Actually, I am not sure this new method works - it is failing to find folder now. I am reviewing now.